File: classifier/data/raw/ai/samples/project_euler_java/Problem100.java
// Project Euler 100: arranged probability with more than one trillion total discs.
public class Problem100 {
    public static void main(String[] args) {
        // x = 2*total-1 and y = 2*blue-1 satisfy x^2 - 2y^2 = -1.
        long x = 7, y = 5;
        do {
            long nextX = 3 * x + 4 * y, nextY = 2 * x + 3 * y;
            x = nextX; y = nextY;
        } while ((x + 1) / 2 <= 1_000_000_000_000L);
        System.out.println((y + 1) / 2);
    }
}

File: classifier/data/raw/human/samples/project_euler_zig/problem_100.zig
const std = @import("std");

// Project Euler 100: first blue-disc count above one trillion total discs.
pub fn main() !void {
    var blue: u64 = 15;
    var total: u64 = 21;
    while (total <= 1_000_000_000_000) {
        const next_blue = 3 * blue + 2 * total - 2;
        total = 4 * blue + 3 * total - 3;
        blue = next_blue;
    }
    var buffer: [32]u8 = undefined;
    try std.Io.File.writeStreamingAll(std.Io.File.stdout(), std.Io.Threaded.global_single_threaded.io(), (try std.fmt.bufPrint(&buffer, "{d}\n", .{blue})));
}
